src/DotNetWorker.OpenTelemetry/ConfigureFunctionsOpenTelemetry.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry;
+using OpenTelemetry.Resources;
+
+namespace Microsoft.Azure.Functions.Worker.OpenTelemetry
+{
+    public static class ConfigureFunctionsOpenTelemetry
+    {
+        private const string DefaultServiceName = "dotnetiso";
+        public static OpenTelemetryBuilder UseFunctionsWorkerDefaults(this OpenTelemetryBuilder builder)
+        {
+            if (builder is null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.Services
+                // Lets the host know that the worker is sending logs to App Insights. The host will now ignore these.
+                .Configure<WorkerOptions>(workerOptions => workerOptions.Capabilities["WorkerOpenTelemetryEnabled"] = bool.TrueString);
+
+            builder.ConfigureResource(r =>
+            {
+                string version = typeof(ConfigureFunctionsOpenTelemetry).Assembly.GetName().Version.ToString();
+                ConfigureResourceAttributes(r, version);
+            });
+
+            return builder;
+        }
+
+        private static string GetServiceName()
+        {
+            return Environment.GetEnvironmentVariable(ResourceAttributeConstants.SiteNameEnvVar) ?? DefaultServiceName;
```

Comment:
I am trying to keep it same for both, host and worker. Some APM vendors group telemetry on service name, I believe it will help if users see both, host and worker telemetry grouped together.

Reply:
If we keep it the same, can we add an attribute to each like `azure.functions.role` `host|worker`